src/neuroconv/tools/testing/data_interface_mixins.py

```diff
@@ -0,0 +1,163 @@
+import json
+from abc import abstractmethod
+from datetime import date, datetime
+from pathlib import Path
+from typing import Callable, Union
+
+import numpy as np
+from jsonschema import validate
+from jsonschema.validators import Draft7Validator
+from roiextractors import NwbImagingExtractor, NwbSegmentationExtractor
+from roiextractors.testing import check_imaging_equal, check_segmentations_equal
+from spikeinterface.core.testing import check_recordings_equal, check_sortings_equal
+from spikeinterface.extractors import NwbRecordingExtractor, NwbSortingExtractor
+
+from ...basedatainterface import BaseDataInterface
+from ...datainterfaces.ecephys.baserecordingextractorinterface import (
+    BaseRecordingExtractorInterface,
+)
+from ...datainterfaces.ecephys.basesortingextractorinterface import (
+    BaseSortingExtractorInterface,
+)
+from ...datainterfaces.ophys.baseimagingextractorinterface import (
+    BaseImagingExtractorInterface,
+)
+from ...datainterfaces.ophys.basesegmentationextractorinterface import (
+    BaseSegmentationExtractorInterface,
+)
+
+
+def json_serial(obj):
+    """JSON serializer for objects not serializable by default json code"""
+
+    if isinstance(obj, (datetime, date)):
+        return obj.isoformat()
+    raise TypeError("Type %s not serializable" % type(obj))
```

Comment:
I'd say just import the `NWBMetaDataEncoder` from `neuroconv.utils` : https://github.com/catalystneuro/neuroconv/blob/main/src/neuroconv/utils/json_schema.py#L16